File: CMBooking-backend/src/main/java/com/booking/CMBookingbackend/service/RentingService.java
package com.booking.CMBookingbackend.service;

import com.booking.CMBookingbackend.dto.PropertyDto;
import com.booking.CMBookingbackend.dto.RentingDto;
import com.booking.CMBookingbackend.response.VerifyResponse;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface RentingService {

    RentingDto createRenting(RentingDto rentingDto);


    List<RentingDto> getAllRentings(String email);

    List<RentingDto> getRentingByPropertyId(Long propertyId);

    void deleteRenting(Long rentingId);
}

File: CMBooking-frontend/src/service/RentingService.js
import axios from "axios";

const REST_API_BASE_URL = 'http://localhost:8080/api/renting'; 

export const createRenting = (renting) => axios.post(REST_API_BASE_URL, renting);

export const getAllRentings = (userEmail) => axios.get(REST_API_BASE_URL + '/' + userEmail);

export const getRentingByPropertyId = (propertyId) => axios.get(REST_API_BASE_URL + '/property/' + propertyId);

export const deleteRenting = (rentingId) => axios.delete(REST_API_BASE_URL + '/' + rentingId);